specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -128,124 +132,126 @@ export function TableUniquenessRules({
     [container.name, loading, tableRules, setTableRules]
   );
 
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const SaveButton = saveBlocked ? Submit.Red : Submit.Save;
```

Comment:
sorry, not quite right yet.

if there is a validation error, then some field on the form should be showing that validation error - using the useValidation hook

if you have validation error, clicking the save button would automatically focus the error message and prevent form submission
the field causing the error would also get red background automatically

please refactor as follows:
1. Use Submit.Save (remove ternary)
2. Get rid of this:

```ts
  const saveBlocked = React.useMemo(
    () =>
      tableRules.some(
        ({ duplicates }) =>
          duplicates !== undefined && duplicates.totalDuplicates !== 0
      ),
    [tableRules]
  );
```

instead, inside of each UniquenessRuleRow (or some sub-component inside of it), do this:

```tsx
const hasDuplicates = fetchedDuplicates !== undefined && fetchedDuplicates.totalDuplicates !== 0;
const { validationRef } = useValidation(hasDuplicates ? schemaText(yourValidationMessageHere));

// and then, on the input field that the user was using to cause this validation error to happen (or input field that is most related to this error):

<Input.Text ... ref={validationRef} />
```

then get rid of the `<H2>{invalidUniqueReason}</H2>`

Reply:
oh, looks like you already resolved this correctly (you are using the useValidation hook correctly)

all good, false alarm - sorry, I got lost in this large file)